service/matching/db_task_manager.go

```diff
@@ -164,14 +179,18 @@ func (d *dbTaskManager) Start() {
 	}
 
 	d.signalDispatch()
-
-	// TODO: consider using goro.Group
-	go d.acquireLoop(context.Background())
-	go d.writerEventLoop(context.Background())
-	go d.readerEventLoop(context.Background())
+	d.group.Go(d.acquireLoop)
+	d.group.Go(d.writerEventLoop)
+	d.group.Go(d.readerEventLoop)
+	<-d.startupChan
 }
 
 func (d *dbTaskManager) Stop() {
+	defer d.group.Wait()
```

Comment:
1. do not block the stop call
2. there is no reason to use goro here, simply close shutdown channel & let the internal coroutine shutdown itself is enough

Reply:
See my comments above